### DISCUSSION TITLE
How to define min and max limit switches for axis?

### DISCUSSION THREAD

**Discussion by ashokmanker** [grblHAL/ESP32#97] [Q&A]
I have searched everything and read documentation for grblhal and tried to look into header files but didn't find how to define min and max limit switches for any axis. I want to define min and max limit switches for x,y,z,a axis

**Comment by terjeio** [comment]
In your own board map? And for which driver? I have not added support for separate max inputs to all drivers yet, some due to limited number of input pins.

**Reply by ashokmanker** [reply]
Thankyou so much for such a quick reply. Yes I am using my own board map. I am using esp32 driver. Assigned mt_det,th1,th2 to max x,y,z and tb to estop and gpio12 to probe in FluidNC. And physically soldered pull up resistor and cap on gpio12 since its picking up noice. Since this pin dont have schottky diode. I have found stepper running much smoother and stepsticks much cooler in grblhal so i want to use grblhal.

**Reply by terjeio** [reply]
[Added now](https://github.com/grblHAL/ESP32/commit/fa7b9196ec64ad6bec013a3ff3d5d29f24ba94e4) - for you to test.

You can use the `$LIM` command to check the pin status.

**Reply by ashokmanker** [reply]
I recompiled the driver with modified boardmap as you mentioned. Now both min and max switches are working. Many many thanks for such a prompt support and feature adding.
